cmd/query/app/static_handler.go

```diff
@@ -27,22 +33,67 @@ const (
 
 var (
 	staticRootFiles = []string{"favicon.ico"}
+	configPattern   = regexp.MustCompile("JAEGER_CONFIG *= *DEFAULT_CONFIG;")
 )
 
 // StaticAssetsHandler handles static assets
 type StaticAssetsHandler struct {
 	staticAssetsRoot string
+	indexHTML        []byte
 }
 
 // NewStaticAssetsHandler returns a StaticAssetsHandler
-func NewStaticAssetsHandler(staticAssetsRoot string) *StaticAssetsHandler {
+func NewStaticAssetsHandler(staticAssetsRoot string, uiConfig string) (*StaticAssetsHandler, error) {
 	if staticAssetsRoot == "" {
 		staticAssetsRoot = defaultStaticAssetsRoot
 	}
 	if !strings.HasSuffix(staticAssetsRoot, "/") {
 		staticAssetsRoot = staticAssetsRoot + "/"
 	}
-	return &StaticAssetsHandler{staticAssetsRoot: staticAssetsRoot}
+	indexBytes, err := ioutil.ReadFile(staticAssetsRoot + "index.html")
+	if err != nil {
+		return nil, errors.Wrap(err, "Cannot read UI static assets")
```

Comment:
Now it fails without UI. Before it was working. Workaround is to create empty `index.html` or do not include static handler if flag is empty 

https://github.com/jaegertracing/jaeger/pull/493 uses query as a docker container, when xdock is executed it builds all images, Now we have to build UI which takes significant time.

Reply:
let's discuss on #497